### PR TITLE
Add size() and isEmpty() methods to indices [ECR-3316]

### PR DESCRIPTION
**Pull request by MakarovS**
## Overview
Add size() and isEmpty() methods to KeySetIndex, ValueSetIndex, MapIndex and ProofMapIndex.

---
See: https://jira.bf.local/browse/ECR-3316

### Definition of Done

- [x] There are no TODOs left in the code
- [x] Change is covered by automated [tests](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#tests)
- [x] The [coding guidelines](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#the-code-style) are followed
- [x] Public API has Javadoc
- [x] Method preconditions are checked and documented in the Javadoc of the method
- [x] Changelog is updated if needed (in case of notable or breaking changes)
- [x] The continuous integration build passes

### CODE REVIEW

File: exonum-java-binding/core/src/main/java/com/exonum/binding/core/storage/indices/KeySetIndexProxy.java [hunk 1]
```diff
@@ -220,6 +220,20 @@ public void remove(E e) {
     nativeRemove(getNativeHandle(), dbElement);
   }
 
+  /**
+   * Returns the number of entries in this set.
```
Comment (dmitry-timofeev) [inline exonum-java-binding/core/src/main/java/com/exonum/binding/core/storage/indices/KeySetIndexProxy.java]: The existing set docs refer to them as 'elements'. Maps, however, use 'entries'.

File: exonum-java-binding/core/src/test/java/com/exonum/binding/core/storage/indices/KeySetIndexProxyIntegrationTest.java [hunk 2]
```diff
@@ -141,6 +142,29 @@ void removeFailsIfSnapshot() {
             K1)));
   }
 
+  @Test
+  void getMapSize() {
```
Comment (dmitry-timofeev) [inline exonum-java-binding/core/src/test/java/com/exonum/binding/core/storage/indices/KeySetIndexProxyIntegrationTest.java]: not map (here and elsewhere)

File: exonum-java-binding/core/src/test/java/com/exonum/binding/core/storage/indices/ProofMapIndexProxyIntegrationTest.java [hunk 3]
```diff
@@ -40,6 +40,7 @@
 import static org.hamcrest.CoreMatchers.notNullValue;
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.core.IsNot.not;
+import static org.junit.Assert.assertEquals;
```
Comment (dmitry-timofeev) [inline exonum-java-binding/core/src/test/java/com/exonum/binding/core/storage/indices/ProofMapIndexProxyIntegrationTest.java]: That's 4th API: please either import the jupiter API or use assertThat(... equalTo())

File: exonum-java-binding/core/src/test/java/com/exonum/binding/core/storage/indices/MapIndexProxyIntegrationTest.java [hunk 4]
```diff
@@ -454,12 +456,25 @@ void clearMultipleItemFork() {
     });
   }
 
+  // TODO: disabled until ECR-3317 completion (here and below)
```
Comment (bullet-tooth) [inline exonum-java-binding/core/src/test/java/com/exonum/binding/core/storage/indices/MapIndexProxyIntegrationTest.java]: `@Disabled` has an argument for this purpose

**Pull request merged**